tethysapp/hydro_var_monitor/controllers.py

```diff
@@ -1,49 +1,125 @@
-from django.http.response import JsonResponse
+import json
+import logging
+import os
 from datetime import date
 from datetime import datetime
-from django.shortcuts import render
-from .ee_auth import *
-import json
 from django.http import JsonResponse, HttpResponseNotAllowed
+from django.http.response import JsonResponse
+from django.shortcuts import render
+from django.urls import path
+# from . import views
 from tethys_sdk.routing import controller
+
 from . import ee_auth
-import logging
-import os
+from .compare import air_temp_compare, precip_compare
+from .ee_auth import *
 from .ee_tools import ERA5, get_tile_url, GLDAS, CHIRPS, IMERG, GLDAS_evapo
 from .plots import plot_ERA5, plot_GLDAS, plot_IMERG, plot_CHIRPS
-from .compare import air_temp_compare, precip_compare
+from .unzip import unzip_and_read_files
+import zipfile
+import tempfile
 
+from .app import HydroVarMonitor as App
 
-@controller(name='home', url='hydro-var-monitor')
-def home(request):
-    if not EE_IS_AUTHORIZED:
-        return render(request, 'hydro_var_monitor/no_auth_error.html')
+
+@controller(name='home', url='hydro-var-monitor', app_workspace=True)
+def home(request, app_workspace):
+    lang = request.GET.get('lang', 'en')
+    if lang not in ['en', 'es']:
+        lang = 'en'
+
+    # if not EE_IS_AUTHORIZED:
+    #     return render(request, 'hydro_var_monitor/no_auth_error.html')
     ee_sources = {
         'air_temp': ['GLDAS', 'ERA5'],
         'precip': ['GLDAS', 'CHIRPS', 'IMERG', 'ERA5'],
         'soil_moisture': ['GLDAS', ],
         'evapo': ['GLDAS']
     }
+    # lang_ops = os.path.join(request, 'hydro_var_monitor/home_language_components')
+    # langs = os.listdir(lang_ops)
+    # list_of_languages = [item for item in langs]
+    list_of_languages = ["check"]
+
+    target_directory = os.path.join(app_workspace.path, "Exact")
+    items = os.listdir(target_directory)
+    list_of_directories = [item for item in items if os.path.isdir(os.path.join(target_directory, item))]
+    if "__MACOSX" in list_of_directories:
+        list_of_directories.remove("__MACOSX")
+    options_for_each_directory = {}
+    for directory in list_of_directories:
+        directory_path = os.path.join(target_directory, directory)
+        files = [name for name in os.listdir(directory_path) if os.path.isfile(os.path.join(directory_path, name))]
+
+        # add the directory and its list of files to the dictionary
+        options_for_each_directory[directory] = files
+
     context = {
-        'sources': json.dumps(ee_sources)
+        'language_options': json.dumps(list_of_languages),
+        'lang': lang,
+        'sources': json.dumps(ee_sources),
+        'option': json.dumps(options_for_each_directory),
+        'directories': json.dumps(list_of_directories)
     }
     return render(request, 'hydro_var_monitor/home.html', context)
 
+
+@controller(name='admin', url='hydro-var-monitor/admin', app_workspace=True)
```

Comment:
this should require the user to be logged in and be admin. See tethys docs for how to use the @login_required decorator and another decorator for being and admin user which i don't remember the name of.